### ISSUE TITLE
[cloud_firestore] [web] [production] "TypeError: firebase.firestore is not a function" when hosting

### ISSUE THREAD

**Issue by blbhargav** [FirebaseExtended/flutterfire#2484]
I am using firestore in flutter web. Everything is working fine in local i.e,Android studio chrome. But when I hosted in firebase hosting (firebase serve) it's not working. The browser console giving error.

```bash
Uncaught TypeError: firebase.firestore is not a function 
at Object.arO (top_level.dart:125) 
at cloud_firestore_web.dart:26 
at alN.a (async_patch.dart:308) 
at alN.$2 (async_patch.dart:333) 
at Object.N (async_patch.dart:238) 
at Object.T8 (main.dart:8) 
at js_helper.dart:3221 
at js_helper.dart:3221 
at dartProgram (js_helper.dart:3221) 
at js_helper.dart:3221 
```

flutter doctor -v

```console
[✓] Flutter (Channel beta, v1.17.0-3.4.pre, on Linux, locale en_IN)
• Flutter version 1.17.0-3.4.pre at /home/bhargav/Android/flutter
• Framework revision e6b34c2 (34 hours ago), 2020-05-02 11:39:18 -0700
• Engine revision 540786dd51
• Dart version 2.8.1

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
• Android SDK at /home/bhargav/Android/Sdk
• Platform android-29, build-tools 29.0.3
• Java binary at: /snap/android-studio/88/android-studio/jre/bin/java
• Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
• All Android licenses accepted.

[✓] Chrome - develop for the web
• Chrome at google-chrome

[✓] Android Studio (version 3.6)
• Android Studio at /snap/android-studio/88/android-studio
• Flutter plugin version 45.1.1
• Dart plugin version 192.7761
• Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] Connected device (3 available)
• Redmi Note 7 Pro • a682ccfc • android-arm64 • Android 9 (API 28)
• Chrome • chrome • web-javascript • Google Chrome 81.0.4044.122
• Web Server • web-server • web-javascript • Flutter Tools

• No issues found!
```

Any help please.
https://stackoverflow.com/q/61584732/9103369

**Comment by iapicca** [comment]
Hi @blbhargav 
can you please provide your
your `pubspec.yaml` 
and a reproducible minimal code sample
Thank you

**Comment by blbhargav** [comment]
pubspec.yaml 
`name: chitragupta
description: Monthly expenditure tacker

version: 1.2.0+2

environment:
  sdk: ">=2.2.2 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^0.1.2
  flutter_launcher_icons: ^0.7.2
  material_design_icons_flutter: ^4.0.5145
  firebase_core: ^0.4.3+2 #For firebase database
  firebase_database: ^3.1.3
  firebase_auth: ^0.16.0 #firebase Authentication
  shared_preferences: ^0.5.7 #for local storage
  shared_preferences_web: ^0.1.2+4
  intl: ^0.16.0
  flutter_datetime_picker: ^1.2.6
  fluttertoast: ^3.1.3
  package_info: ^0.4.0+13
  image_picker: ^0.6.2+3
  http: ^0.12.0+2
  charts_flutter: ^0.8.1
  flutter_svg: ^0.17.4
  cloud_firestore: ^0.13.5


dev_dependencies:
  flutter_test:
    sdk: flutter


# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  uses-material-design: true
  assets:
    - assets/

flutter_icons:
  image_path: "assets/logo.png"
  android: true
  ios: true
`
Well the app working perfectly fine in chrome (flutter run -d chrome). But not in local server mode or in firebase hosting

**Comment by emillenard** [comment]
@blbhargav try to remove "defer" if you have it in index.html.
replace: "<script defer src="https://www.gstatic.com/firebasejs/7.14.5/firebase-firestore.js"></script> 
with "<script src="https://www.gstatic.com/firebasejs/7.14.5/firebase-firestore.js"></script>"

**Comment by kevan-mtl** [comment]
hi there,

I get exactly the same issue:

`[✓] Flutter (Channel master, 1.18.0-10.0.pre, on Mac OS X 10.14.6 18G4032, locale en-CA)
    • Flutter version 1.18.0-10.0.pre at /Users/kevansaba/flutter
    • Framework revision 6a75dc449e (5 weeks ago), 2020-05-01 13:04:01 -0700
    • Engine revision 2db3276573
    • Dart version 2.9.0 (build 2.9.0-3.0.dev 726d3c7725)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/kevansaba/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 43.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.45.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.2

[✓] Connected device (2 available)
    • Web Server • web-server • web-javascript • Flutter Tools
    • Chrome     • chrome     • web-javascript • Google Chrome 83.0.4103.61

• No issues found!`

My pubspec.yaml looks like this:
`name: flutterwebtest
description: A new Flutter project.

publish_to: 'none'

version: 1.0.0+1

environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^0.1.3
  get_it: ^4.0.2
  responsive_builder: ^0.1.4
  flutter_bloc: ^4.0.0
  firebase_auth: ^0.16.0
  equatable: ^1.1.1
  provider: ^4.0.5
  flutter_spinkit: ^4.1.2+1
  time_machine: ^0.9.12
  table_calendar: ^2.2.3
  searchable_dropdown: ^1.1.3

# COMMENTING OUT FLUTTER TEST 
dev_dependencies:
  flutter_test:
    sdk: flutter

flutter:
  uses-material-design: true`

Like Bhargav, I can make the app run using flutter run, but it won't work with Firebase hosting (neither locally or hosted remotely). 

Lastly, this is my web/index.html file:

`<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <meta content="IE=Edge" http-equiv="X-UA-Compatible">
  <meta name="description" content="A new Flutter project.">

  
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  <meta name="apple-mobile-web-app-title" content="flutterwebtest">
  <link rel="apple-touch-icon" href="icons/Icon-192.png">

  
  <link rel="shortcut icon" type="image/png" href="favicon.png"/>

  <title>flutterwebtest</title>
  <link rel="manifest" href="manifest.json">
</head>
<body>
  
  <script>
    if ('serviceWorker' in navigator) {
      window.addEventListener('load', function () {
        navigator.serviceWorker.register('flutter_service_worker.js');
      });
    }
  </script>



<script src="/__/firebase/7.15.0/firebase-app.js"></script>


<script src="/__/firebase/7.15.0/firebase-analytics.js"></script>


<script src="/__/firebase/init.js"></script>

  <script src="main.dart.js" type="application/javascript"></script>
</body>
</html>
`

Any ideas?

**Comment by shajidhasan** [comment]
Having the same problem...

**Comment by TimWhiting** [comment]
Same here, works if I load the page without the /#/ at the end of the url, but then when refreshing obviously the /#/ is on the route already, so I get a white page with this error.

**Comment by Buildex** [comment]
Same error here.

**Comment by AseemWangoo** [comment]
> @blbhargav try to remove "defer" if you have it in index.html.
> replace: "<script defer src="https://www.gstatic.com/firebasejs/7.14.5/firebase-firestore.js"></script>
> with "<script src="https://www.gstatic.com/firebasejs/7.14.5/firebase-firestore.js"></script>"

i solved using this

**Comment by Salakar** [comment]
Hey all, as part of our on-going work for #2582, this has been resolved in our Firebase Firestore rework (#2913) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
